cylc/flow/cfgspec/workflow.py

```diff
@@ -996,6 +996,144 @@ def get_script_common_text(this: str, example: Optional[str] = None):
+
+                The ``completion`` configuration allows you to override the
+                default completion to suit your needs.
+
+                E.g., in this example, the task ``foo`` has three optional
+                outputs, ``x``, ``y`` and ``z``:
+
+                .. code-block:: cylc-graph
+
+                   foo:x? => x
+                   foo:y? => y
+                   foo:z? => z
+                   x | y | z => bar
+
+                Because all three of these outputs are optional, if none of
+                them are generated, the task's outputs will still be
+                considered complete.
+
+                If you wanted to require that at least one of these outputs is
+                generated you can configure the completion condition like so:
+
+                .. code-block:: python
+
+                   # the task must succeed and generate at least one of the
+                   # outputs "x" or "y" or "z":
+                   succeeded and (x or y or z)
+
+                .. note::
+
+                   For the completion expression, hyphens in task outputs
+                   are converted into underscores to allow
+                   evaluation by Python, e.g.:
+
+                   .. code-block:: cylc
+
+                      [runtime]
+                          [[foo]]
+                              completion = succeeded and my_output # underscore
+                              [[[outputs]]]
+                                  my-output = 'my custom task output' # hyphen
+
+                .. note::
+
+                   In some cases the succeeded output might not explicitly
+                   appear in the graph, e.g:
+
+                   .. code-block:: cylc-graph
+
+                      foo:x? => x
+
+                   In these cases succeess is presumed to be required unless
+                   explicitly stated otherwise, either in the graph:
+
+                   .. code-block:: cylc-graph
+
+                      foo?
+                      foo:x? => x
+
+                   Or in the completion expression:
+
+                   .. code-block:: cylc
+
+                      completion = x  # no reference to succeeded here
+
+
+                .. hint::
+
+                   If task outputs are optional in the graph they must also
+                   be optional in the completion condition and vice versa.
+
+                   .. code-block:: cylc
+
+                      [scheduling]
+                          [[graph]]
+                              R1 = """
+                                  # ERROR: this should be "a? => b"
+                                  a => b
+                              """
+                      [runtime]
+                          [[a]]
+                              # this completion condition implies that the
+                              # succeeded output is optional
+                              completion = succeeded or failed
```

Comment:
Maybe put the runtime section above the graph for clarity

Reply:
We always put `[scheduling]` before `[runtime]`.

Reply:
Maybe split it up, as I think seeing
```
# ERROR: this should be "a? => b"
a => b
```
*before* seeing why it is an error may be confusing.

Perhaps:
```suggestion
                      [scheduling]
                          [[graph]]
                              R1 = a => b
                      [runtime]
                          [[a]]
                              # this completion condition means that the
                              # succeeded output is optional
                              completion = succeeded or failed
                              
                   In the example above, the graph could be changed to `a? => b`
                   in order to comply with the completion statement.
```

Reply:
I don't see an issue with this.

Reply:
I've updated the suggestion, let me know what you think

Reply:
Still prefer the original TBH but I'll elaborate on the comments a bit if you think it is necessary.